### ISSUE TITLE
Create signalregister wrapper

### ISSUE THREAD

**Issue by hellow554** [Kistler-Group/sdbus-cpp#149]
This "bug" costed me more than one day of hard thinking... :facepalm:

I have a sdbus client and I want to wait for two signals, so I used `uponSignal`. The convinient API really makes it hard to misuse, but not impossible, because I forgot `proxy->finishRegistration()`.

I would love to see something like this (just a draft):

```
proxy->register_signals(
    sdbus::createSignal("com.example.iface", "Completed", [](int result) { fmt::print("I got {}\n", result); ),
    sdbus::createSignal("org.freedesktop.DBus.Properties", "PropertiesChanged", propchange_callback),
);
```

where in the end the `register_signals` function will call `finishRegistration` on its own.

**Comment by sangelovic** [comment]
Hey @hellow554. I have finally found time to look into this, and within my ongoing work on `release/v2.0` feature branch, I came with a proposal of a new API that hopefully saves other users from days of hard thinking :-)

The new API is similar to what you've proposed, but also aligns more with the API of the underlying sd-bus library, while also aiming to keep readability and simplicity. The user now registers at once the whole vtable for an object interface:

```c++
object_->addVTable( sdbus::setInterfaceFlags().markAsDeprecated().withPropertyUpdateBehavior(sdbus::Flags::EMITS_NO_SIGNAL)
                  , sdbus::registerMethod("multiply").withInputParamNames("a", "b").withOutputParamNames("result").implementedAs([this](const int64_t& a, const double& b){ return this->multiply(a, b); })
                  , sdbus::registerSignal("signalWithMap").withParameters<std::map<int32_t, std::string>>("aMap").markAsDeprecated()
                  , sdbus::registerProperty("action").withGetter([this](){ return this->action(); }).withSetter([this](const uint32_t& value){ this->action(value); }).withUpdateBehavior(sdbus::Flags::EMITS_INVALIDATION_SIGNAL)
                  ).forInterface(INTERFACE_NAME);
```

The registration gets active right after the call. This mimics the behavior of [sd_bus_add_object_vtable()](https://manpages.ubuntu.com/manpages/focal/en/man3/sd_bus_add_object_vtable.3.html), so we are more consistent with the design of sd-bus API, but on higher abstraction layer.

Most importantly, no extra `finishRegistration()` step is needed anymore ;-)

There is an open WIP PR https://github.com/Kistler-Group/sdbus-cpp/pull/388 with that new API.

Ad.: Similarly, on the `Proxy` side, for consistency and simplicity, signals will be registered right after the `registerSignalHandler()` method call, and no `finishRegistration()` is needed anymore. (An open WIP PR for this is https://github.com/Kistler-Group/sdbus-cpp/pull/389).

What do you think about it?

**Comment by hellow554** [comment]
> Most importantly, no extra finishRegistration() step is needed anymore ;-)

🎉 🎉 🎉 

Amazin. It happened so often to me, you won't believe it.

I can see that sticking as close as possible to the vtable api spares you some headaches, so that's totally fine IMHO.
Way to go!

**Comment by sangelovic** [comment]
Solved as part of #364 which is in the upcoming days going to be released as `v2.0.0`
